testdata/deep_exit.go

```diff
@@ -36,3 +37,15 @@ func TestMain(m *testing.M) {
 	// must match because this is not a test file
 	os.Exit(m.Run()) // MATCH /calls to os.Exit only in main() or init() functions/
 }
+
+func flagParseOutsideMain() {
+	flag.Parse() // MATCH /calls to flag.Parse only in main() or init() functions/
+}
+
+func flagNewFlagSetExitOnErrorOutsideMain() {
+	flag.NewFlagSet("cmd", flag.ExitOnError) // MATCH /calls to flag.NewFlagSet only in main() or init() functions/
```

Comment:
As I understand from the rule implementation and test, not all calls to `flag.NewFlagSet` are problematic. Only those that have `flag.ExitOnError` as argument are. If my understanding is correct then the failure message is not precise enough.